scipy/spatial/transform/tests/test_rotation.py

```diff
@@ -640,6 +665,55 @@ def test_magnitude_single_rotation():
     assert_allclose(result2, 0)
 
 
+def test_reduction_no_indices():
+    result = Rotation.identity().reduce(return_indices=False)
+    assert type(result) == Rotation
+
+
+def test_reduction_none_indices():
+    result = Rotation.identity().reduce(return_indices=True)
+    assert type(result) == tuple
+    assert len(result) == 3
+
+    reduced, left_best, right_best = result
+    assert left_best is None
+    assert right_best is None
+
+
+def test_reduction_full_indices():
+    rng = np.random.RandomState(0)
+    l = Rotation.random(5, random_state=rng)
+    r = Rotation.random(10, random_state=rng)
+    p = Rotation.random(7, random_state=rng)
+    result = p.reduce(l, r, return_indices=True)
+    assert type(result) == tuple
+    assert len(result) == 3
+
+    reduced, left_best, right_best = result
+    assert len(left_best) == len(p)
+    assert len(right_best) == len(p)
+
+
+def test_reduction_scalar_calculation():
+    rng = np.random.RandomState(0)
+    l = Rotation.random(5, random_state=rng)
+    r = Rotation.random(10, random_state=rng)
+    p = Rotation.random(7, random_state=rng)
+    reduced, left_best, right_best = p.reduce(l, r, return_indices=True)
+
+    # Loop implementation of the vectorized calculation in Rotation.reduce
+    scalars = np.zeros((len(l), len(p), len(r)))
+    for i, li in enumerate(l):
+        for j, pj in enumerate(p):
+            for k, rk in enumerate(r):
+                scalars[i, j, k] = np.abs((li * pj * rk).as_quat()[3])
+    scalars = np.reshape(np.rollaxis(scalars, 1), (scalars.shape[1], -1))
+
+    max_ind = np.argmax(np.reshape(scalars, (len(p), -1)), axis=1)
+    assert (left_best == max_ind // len(r)).all()
+    assert (right_best == max_ind % len(r)).all()
```

Comment:
Let's also test that reduced rotations by both methods are identical (very close).

Reply:
Good suggestion.  I have added it.
To avoid dealing with the quaternion sign ambiguity I have used
```
mag = (reduced.inv() * reduced_check).magnitude()
```
I would like to implement an "angular_distance" (or "misorientation") convenience function which contains this logic. It would make for better readability. See below for feature suggestions.